pkg/kubelet/certificate/bootstrap/bootstrap.go

```diff
@@ -246,6 +247,14 @@ func isClientConfigStillValid(kubeconfigPath string) (bool, error) {
 		utilruntime.HandleError(fmt.Errorf("unable to load transport configuration from existing bootstrap client config read from %s: %v", kubeconfigPath, err))
 		return false, nil
 	}
+	// prevent TLSConfigFor from getting mad about invalid dialer config with TLS reloading
+	// this is okay because we do not plan on using this TLS config
+	transportConfig.TLS.CallerHandlesDialer = true
+	if transportConfig.DialHolder == nil {
```

Comment:
If we assume our understanding of this code will only degrade over time, that means this moment is the best we will ever understand this. That doesn't bode well...